tests/unit/settled-test.js

```diff
@@ -0,0 +1,270 @@
+          }, 25);
+        },
+
+        willDestroy() {
+          this._super.apply(this, arguments);
+          // must be called with `Ember.Test` as context for Ember < 2.8
+          Ember.Test.unregisterWaiter(this, this.isReady);
+        },
+      })
+    );
+
+    owner.register('template:components/x-test-5', hbs`{{internalValue}}`);
+
+    this.server = new Pretender(function() {
+      this.get(
+        '/whazzits',
+        function() {
+          return [200, { 'Content-Type': 'text/plain' }, 'Remote Data!'];
+        },
+        25
+      );
+    });
+  });
+
+  hooks.afterEach(async function() {
+    this.server.shutdown();
+    await settled();
+
+    teardownContext(this);
+    teardownRenderingContext(this);
+  });
+
+  test('it works when async exists in `init`', async function(assert) {
+    await this.render(hbs`{{x-test-1}}`);
+
+    await settled();
+
+    assert.equal(this.element.textContent, 'async value');
+  });
+
+  test('it works when async exists in an event/action', async function(assert) {
+    await this.render(hbs`{{x-test-2}}`);
+
+    assert.equal(this.element.textContent, 'initial value');
+
+    fireEvent(this.element.querySelector('div'), 'click');
+
+    await settled();
+
+    assert.equal(this.element.textContent, 'async value');
+  });
+
+  test('it waits for AJAX requests to finish', async function(assert) {
+    await this.render(hbs`{{x-test-3}}`);
+
+    fireEvent(this.element.querySelector('div'), 'click');
+
+    await settled();
+
+    assert.equal(this.element.textContent, 'Remote Data!');
+  });
+
+  test('it waits for interleaved AJAX and run loops to finish', async function(assert) {
+    var testContext = this;
+
+    await this.render(hbs`{{x-test-4}}`);
+
+    fireEvent(this.element.querySelector('div'), 'click');
+
+    await settled();
+
+    assert.equal(testContext.element.textContent, 'Local Data!Remote Data!Remote Data!');
+  });
+
+  test('it can wait only for AJAX', async function(assert) {
+    var testContext = this;
+
+    await this.render(hbs`{{x-test-4}}`);
+
+    fireEvent(this.element.querySelector('div'), 'click');
+
+    await settled({ waitForTimers: false });
+
+    assert.equal(testContext.element.textContent, 'Local Data!Remote Data!');
+  });
+
+  // in the wait utility we specific listen for artificial jQuery events
+  // to start/stop waiting, but when using ember-fetch those events are not
+  // emitted and instead test waiters are used
+  //
+  // therefore, this test is only valid when using jQuery.ajax
+  (hasjQuery() ? test : skip)('it can wait only for timers', async function(assert) {
+    var testContext = this;
+
+    await this.render(hbs`{{x-test-4}}`);
+
+    fireEvent(this.element.querySelector('div'), 'click');
+
+    await settled({ waitForAJAX: false });
+
+    assert.equal(testContext.element.textContent, 'Local Data!');
+  });
+
+  test('it waits for Ember test waiters', async function(assert) {
+    await this.render(hbs`{{x-test-5}}`);
```

Comment:
since `render()` without `this` is apparently meant to be the documented default should we use it here too?

Reply:
Ya, but I didn’t want to do that here since I wasn’t sure how you felt about the other PR yet.